docs/release.md

```diff
@@ -0,0 +1,180 @@
+
+2. Git tag release (`vn.x`)
+3. Build openshift-azure container image (`sync:vn.x`, etc)
+4. If the minor release is a cherry-pick (not a straightforward branch of master), carry out update testing before release to MSFT
+5. (TBD) CI test on the result? User documentation release procedure? Release procedure to MSFT?
+
+Branching model
+```
+-------master--*----------*---------------------->
+               \          \
+                \           \-release-v4----T(v4.0)-----T(v4.1)->
+                 \-release-v3------T(v3.0)---->
+```
+
+
+# Doing a release with current integration:
+
+1. Create a branch from master and push it. This will create branch for us to do release in.
+
+```
+git checkout <required commit> 
+git checkout -b release-vx
+# we create release branch for first weekly only
+git push upstream release-vx
+```
+
+If it is minor release:
+```
+git checkout release-vx
+git checkout -b release-vx.y-fix
+git cherry-pick <git commit id>
+```
+
+Open a PR into release branch.
+
+2. Configure testing for release branch in `openshift/release` repository
+
+```
+git checkout master
+git fetch upstream 
+git rebase upstream/master 
+git checkout -b osa.vx.y.release
+```
+
+Add new branch config file in `ci-operator/config/openshift/openshift-azure/openshift-openshift-azure-release-vx.yaml`
+
+Add new test jobs for release branch:
+`ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-release-vx-presubmits.yaml`
+
+Run prowgen to validate configuration:
+```
+docker pull registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest
+docker run -it -v $(pwd)/ci-operator:/ci-operator:z registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest --from-dir /ci-operator/config/ --to-dir /ci-operator/jobs
+```
+
+Merge it to release repository. This will make sure that any PR to release branches is gates by tests.
+
+3. Create release PR into `openshift/openshift-azure` repository release branch:
+
+```
+git checkout upstream/release-vx
+git checkout -b release-vx-pluginconfig
+```
+
+update `pluginconfig/pluginconfig-311.yaml` file with image version for this particular release
+```
+clusterVersion: vx.y
+imageVersion: 311.69.20190214
+images:
+  ansibleServiceBroker: registry.access.redhat.com/openshift3/ose-ansible-service-broker:v3.11.69
+  azureControllers: quay.io/openshift-on-azure/azure-controllers:vx.y
+  etcdBackup: quay.io/openshift-on-azure/etcdbackup:vx.y
+```
+
+Make sure you update `clusterVersion` field all image version to point to specific version, instead of `latest` tag.
+
+Merge this PR into `release-vx` branch. Test should be using test infrastructure built images.
+
+4. Tag release for the release you just merged in step 3.
+
+This step requires elevated right on git repo. You might need to ask somebody else to execute these commands.
+
+```
+git checkout upstream/release-vx
+git tag -a vx.y -m 'reason' # where reason is release summary in one sentence
```

Comment:
git tag -a -s -m "Version vx.y" vx.y  # Better to sign the release, Tagger needs a GPG key setup in github